ClientCore/I18N/Translation.cs

```diff
@@ -0,0 +1,302 @@
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using ClientCore.Extensions;
+using Rampastring.Tools;
+using Rampastring.XNAUI.XNAControls;
+
+namespace ClientCore.I18N;
+
+public class Translation : ICloneable
+{
+    public static Translation Instance { get; set; }
+
+    /// <summary>The translation metadata section name.</summary>
+    public const string METADATA_SECTION = "General";
+
+    public string LocaleCode { get; private set; } = string.Empty;
+
+    /// <summary>The explicitly set UI name for the translation.</summary>
+    private string _name = string.Empty;
+    /// <summary>The UI name for the translation.</summary>
+    public string Name
+    {
+        get => string.IsNullOrWhiteSpace(_name) ? GetLanguageName(LocaleCode) : _name;
+        private set => _name = value;
+    }
+
+    /// <summary>Shows the information about the author.</summary>
+    public string Author { get; private set; } = string.Empty;
+
+    /// <summary>
+    /// The key stands for a key name, and the value stands for a string that is used in System.string.Format().
+    /// The value can not contain IniNewLinePattern when loading or saving via ini format.
+    /// </summary>
+    private Dictionary<string, string> Values { get; } = new();
+
+    // public bool IsRightToLeft { get; set; } // TODO
+
+    /// <summary>Contains all keys with missing translations.</summary>
+    private readonly HashSet<string> MissingKeys = new HashSet<string>();
+
+    /// <summary>Used to write missing translation table entries to a file.</summary>
+    public const string MISSING_KEY_PREFIX = "; ";  // a hack but hey it works
+
+    /// <summary>Used for hardcoded strings.</summary>
+    public const string CLIENT_PREFIX = "Client";
+    /// <summary>Used for INI values.</summary>
+    public const string INI_PREFIX = "INI";
+    /// <summary>Used for INI values.</summary>
+    public const string CONTROLS_PREFIX = "Controls";
+    /// <summary>Used for parent-agnostic INI values.</summary>
+    public const string GLOBAL_PREFIX = "Global";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Translation"/> class.
+    /// </summary>
+    public Translation(string translationCode)
+    {
+        if (string.IsNullOrWhiteSpace(translationCode))
+            throw new ArgumentException($"'{nameof(translationCode)}' cannot be null or whitespace.", nameof(translationCode));
+
+        LocaleCode = translationCode;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Translation"/> class
+    /// that loads the translation from an ini file.
+    /// </summary>
+    /// <param name="ini">An ini file to be read.</param>
+    /// <param name="localeCode">A locale code for this translation.</param>
+    public Translation(IniFile ini, string localeCode)
+    {
+        LocaleCode = localeCode;
+
+        if (ini == null)
+            throw new ArgumentNullException(nameof(ini));
```

Comment:
Dropping legacy .NET 4 support enabling us to use `ArgumentNullException.ThrowIfNull(ini)` to shorten the code